### ISSUE TITLE
Transitive analyzers cause conflicts when strict deps are disabled

### ISSUE THREAD

**Issue by sin-ack** [bazelbuild/rules_dotnet#443]
There is a simplistic check to elide assemblies when they are already present, but the same isn't done for analyzers:
https://github.com/bazelbuild/rules_dotnet/blob/626e28effb14f87de8375186ebb59911a16b71fb/dotnet/private/common.bzl#L315

I ended up having the following compile flags passed down:
```
/analyzer:external/_main~main_extension~nuget.microsoft.extensions.configuration.binder.v8.0.1/analyzers/dotnet/cs/Microsoft.Extensions.Configuration.Binder.SourceGeneration.dll
...
/analyzer:external/rules_dotnet~~targeting_packs_extension~nuget.microsoft.aspnetcore.app.ref.v8.0.3/analyzers/dotnet/cs/Microsoft.Extensions.Configuration.Binder.SourceGeneration.dll
```

Which gives me the following compile error:
```
Microsoft.Extensions.Configuration.Binder.SourceGeneration/Microsoft.Extensions.Configuration.Binder.SourceGeneration.ConfigurationBindingGenerator/BindingExtensions.g.cs(40,10): error CS0433: The type 'InterceptsLocationAttribute' exists in both 'Server, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' and 'Server, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'
Microsoft.Extensions.Configuration.Binder.SourceGeneration/Microsoft.Extensions.Configuration.Binder.SourceGeneration.ConfigurationBindingGenerator/BindingExtensions.g.cs(40,10): error CS0433: The type 'InterceptsLocationAttribute' exists in both 'Server, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' and 'Server, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'
```

Should the same transitive check be added for analyzers?

**Comment by purkhusid** [comment]
Any chance that you could create a small repro of this? Also, does this work with strict deps enabled?

**Comment by sin-ack** [comment]
Sorry for not responding. I'll try to create a repro in the weekend. Yes, this works fine with strict deps as analyzers aren't transitively included then. I'd be interested in figuring out if there's a well-defined assembly resolution algorithm so that we can have a general way of solving this.

**Comment by nvborisenko** [comment]
@purkhusid  I am facing the same behaviour:
```
System.Text.Json.SourceGeneration\System.Text.Json.SourceGeneration.JsonSourceGenerator\SeleniumManagerSerializerContext.PropertyNames.g.cs(18,74): error CS0102: The type 'SeleniumManagerSerializerContext' already contains a definition for 'PropName_browser_path'
System.Text.Json.SourceGeneration\System.Text.Json.SourceGeneration.JsonSourceGenerator\SeleniumManagerSerializerContext.PropertyNames.g.cs(18,74): error CS0102: The type 'SeleniumManagerSerializerContext' already contains a definition for 'PropName_browser_path'
Target //dotnet/src/webdriver:webdriver-netstandard2.0 failed to build
```

This is BUILD
```
csharp_library(
    target_frameworks = [
        "netstandard2.0",
    ],
    deps = [
        framework("nuget", "NETStandard.Library"),
        framework("nuget", "System.Memory"),
        framework("nuget", "System.Text.Json"),
    ],
)
```

Interesting, that it works good with .net8 as target framework:
```
csharp_library(
    target_frameworks = [
        "net8.0",
    ],
    deps = [
    ],
)
```

We don't define dependencies here because of `System.Test.Json` is a part of .net8 sdk.

**Comment by purkhusid** [comment]
I've got a probable fix for this on this branch: https://github.com/bazelbuild/rules_dotnet/pull/452
Would appreciate if you could try this out in your projects.

**Comment by purkhusid** [comment]
Fixed by #452
